src/main/kotlin/no/nav/familie/ef/sak/vilkår/regler/evalutation/RegelEvaluering.kt

```diff
@@ -42,6 +42,7 @@ object RegelEvaluering {
     fun utledVilkårResultat(delvilkårResultat: Map<RegelId, Vilkårsresultat>): Vilkårsresultat {
         return when {
             delvilkårResultat.values.all { it == Vilkårsresultat.OPPFYLT } -> Vilkårsresultat.OPPFYLT
+            delvilkårResultat.values.all { it == Vilkårsresultat.AUTOMATISK_OPPFYLT } -> Vilkårsresultat.AUTOMATISK_OPPFYLT
```

Comment:
Betyr ikke dette at vi setter `AUTOMATISK_OPPFYLT` på topp-vilkårsnivå hvis alle delvilkår er `AUTOMATISK_OPPFYLT` ? 🤔

Reply:
Lag gjerne noen tester på dette også. 
Nå forutsetter vi forresten at vi aldri har et vilkår som har to delvilkår hvor ett er automatisk oppfylt og et annet er oppfylt. Det vil isåfall falle igjennom til error

Reply:
Jeg tror dette burde bli riktig:

```
delvilkårResultat.values.all { it == Vilkårsresultat.OPPFYLT ||  it == Vilkårsresultat.AUTOMATISK_OPPFYLT} -> Vilkårsresultat.OPPFYLT
delvilkårResultat.values.all { it == Vilkårsresultat.OPPFYLT || it == Vilkårsresultat.IKKE_OPPFYLT ||  it == Vilkårsresultat.AUTOMATISK_OPPFYLT} ->
                Vilkårsresultat.IKKE_OPPFYLT
            
```

Reply:
Enig, la til den forbedringen. Grunnen til at dette virker uten den koden er fordi vi har metoden utledResultat (RegelEvaluering:31) som gjør om til OPPFYLT (eller IKKE_OPPFYLT). Jeg la til en test som dokumenterer dette i RegelEvalueringTest. DTO'n som sendes til frontend vil fortsatt inneholde AUTOMATISK_OPPFYLT i delvilkåret, slik at det blir seende riktig ut. 

Synes ikke det var så intuitivt å skjønne sammenhengen når jeg leser koden, og lurer på om det kunne vært lurt å skrive om litt, men tror det blir såpass med endringer at det burde holdes utenfor denne PR'n.

Jeg gjør en ny test i preprod, og passer på at Mirja får testet på denne branchen før det merges og prodsettes, for å være mer sikker på at det virker som det skal.